source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -273,6 +276,32 @@ VhRateLimitOptions Filter::getVirtualHostRateLimitOption(const Router::RouteCons
   return vh_rate_limits_;
 }
 
+const Router::RateLimitPolicy&
+Filter::getRateLimitPolicy(const Router::RouteConstSharedPtr& route,
```

Comment:
I suggest splitting this into getRouteRateLimitPolicy and getVHRateLimitPolicy and getting rid of the new enum. There's almost no common code in the method for the different enum values